src/app/CommandPathParams.h

```diff
@@ -19,6 +19,7 @@
 #pragma once
 
 #include <app/util/basic-types.h>
+#include <lib/core/GroupId.h>
```

Comment:
Why?  basic-types.h includes `lib/core/DataModelTypes.h` which includes `lib/core/GroupId.h`.

Including it directly here does not hurt, but why was it needed?

Reply:
It is not needed for compilation so "missing" in the header isn't quite right really.

Reply:
@bzbarsky-apple - we had the same comment from @mrjerryjohns . The reason is "include what you use" checkers that force not to rely on inter-include guarantees that may change in time.

So agree not needed for compiles, however it makes some cleanliness sense (this is why include what you use checkers exist) and it makes some users of the SDK have an easier life to integrate.

Reply:
https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md is the rationale I believe.

Reply:
I updated summary based on my understanding of the patch: it is not that the include is missing (it is not strictly needed) but that we want to IWYU.

Reply:
Yes, from an IWYU pov this makes total sense.  It does mean headers like DataModelTypes.h are not that useful... ;)